src/linear_rsolve.jl

```diff
@@ -108,8 +108,8 @@ function rsolve!(solver::RLSSolver, A, b, x::AbstractVector)
     while check_stop_criterion(solver.log, solver.stop) == false
         iter += 1
         samp = sample(solver.sampler, A, b, x, iter)
-        rsubsolve!(solver.routine, x, samp, iter)
         log_update!(solver.log, solver.sampler, x, samp, iter, A, b)
+        rsubsolve!(solver.routine, x, samp, iter)
```

Comment:
@npritch928 why was this needed?

Reply:
If we use sketch $(\tilde A_k,$\tilde b_k)$ the update would be $x_k = x_{k-1} - \tilde A_k^\top (\tilde A_k \tilde A_k)^\dagger (\tilde A_k x_{k-1} - \tilde b_k)$, if we compute the sketched residual of that value using the same sketch, which would have happened when rsubsolve is before log_update, this residual will always be zero. $\tilde A_k x_{k-1} - \tilde A_k \tilde A_k^\top (\tilde A_k \tilde A_k)^\dagger (\tilde A_k x_{k-1} - \tilde b_k) - \tilde b_k = 0$. So the order needed to be changed